recipes/tesseract/all/conanfile.py

```diff
@@ -31,50 +47,54 @@ def source(self):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.options.shared:
+            del self.options.fPIC
         if self.options.with_training:
             # do not enforce failure and allow user to build with system cairo, pango, fontconfig
             self.output.warn("*** Build with training is not yet supported, continue on your own")
 
     def configure(self):
-        # Exclude old compilers not supported by tesseract
-        compiler_version = tools.Version(self.settings.compiler.version)
-        if (self.settings.compiler == "gcc" and compiler_version < "5") or \
-                (self.settings.compiler == "clang" and compiler_version < "5"):
-            raise ConanInvalidConfiguration("tesseract/{} requires Clang >= 5".format(self.version))
+        compiler = str(self.settings.compiler)
+        compiler_version = Version(self.settings.compiler.version.value)
+
+        minimal_version = {
+            "Visual Studio": "14",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "6"
+        }
+
+        if compiler in minimal_version and \
```

Comment:
typically we link to give the consumer a warning.

https://github.com/conan-io/conan/issues/8002

Reply:
Raising exceptions is required for dropping unsupported compilers which would fail a build. Did you mean a check like `if compiler not in minimal_version:`?

Reply:
Yes! That's what I meant

```py
if not compiler in minimal_version:
    self.output.warn(...)
else if compiler_version < minimal_version[compiler]:
    raise
```